src/python/visclaw/ipyclaw.py

```diff
@@ -6,15 +6,23 @@
 """
 
 from __future__ import absolute_import
-import sys
-import clawpack.visclaw.Iplotclaw as Iplotclaw
+from argparse import ArgumentParser
+from clawpack.visclaw.Iplotclaw import Iplotclaw
 
-def run_iplotclaw(setplot='setplot.py',outdir='./_output'):
-    ip = Iplotclaw.Iplotclaw(setplot=setplot,outdir=outdir)
+def run_iplotclaw(**kwargs):
+    ip = Iplotclaw(**kwargs)
     ip.plotloop()
 
+def parse_args():  # TODO: ipython cannot parse custom arguments
+    if 'get_ipython' in locals():
+        raise Exception(f"Using ipython, arguments cannot be parsed.")
+    parser = ArgumentParser()
+    parser.add_argument("--setplot", default="setplot.py", type=str, required=False)
+    parser.add_argument("--outdir", default="./_output", type=str, required=False)
+    parser.add_argument("--fname", default=None, type=str, required=False)
+    parser.add_argument("--fps", default=None, type=float, required=False)
+    args = parser.parse_args()
+    return args
+
 if __name__ == '__main__':
-    if len(sys.argv) > 1:
-        run_iplotclaw(*sys.argv[1:])
-    else:
-        run_iplotclaw()
+    run_iplotclaw(setplot="setplot.py", outdir="./_output")
```

Comment:
Is this just a placeholder until the command line parsing can be used?

Reply:
Yes, `**parse_args().__dict__` is now passed instead.